webserver/webserver/src/main/java/io/helidon/webserver/ServerBasicConfig.java

```diff
@@ -210,6 +218,7 @@ static class SocketConfig implements SocketConfiguration {
         private final BackpressureStrategy backpressureStrategy;
         private final int maxUpgradeContentLength;
         private List<RequestedUriDiscoveryType> requestedUriDiscoveryTypes;
+        private Whitelist trustedProxies;
```

Comment:
@tomas-langer I wanted to clarify part of your comment.

What config setting did you have in mind for selecting "trust all" or "trust only allow-listed" behavior? 

If the user configures `server.requested-uri-discovery` at all then by default the discovery behavior ought to be  enabled and governed by whatever other settings are assigned within that config section. (It can be disabled using `requested-uri-discovery.enabled=false` or by removing the `requested-uri-discovery` config section entirely.)

The user can already select "trust all" like this:
```
server:
  requested-uri-discovery:
    types: forwarded
    trusted-proxies:
      allow:
        all: true
```
If the user has a `requested-uri-discovery` setting then isn't that by itself selecting `trust only allow-listed`?

Reply:
I think with our subsequent discussions we are in agreement on what combinations of settings are valid. Here is what the code as written (I'm about to push the latest) does and is a candidate for what we could add to the doc for the webserver to describe this feature: (I've written in terms of config but the same applies to settings assigned using the builder):

In `SocketConfiguration.Builder` (which means immediately under `server` or under one of the `server.sockets` entries):

* `requested-uri-discovery` is optional.
* If `requested-uri-discovery` is disabled (either because it is absent or it is present and `requested-uri-discovery.enabled` is set to `false`) Helidon:
   *  ignores any `Forwarded` and `X-Forwarded` headers and adopts the `HOST` discovery type, meaning the `Host` header is used for host info and the request URI's scheme and port are used;
   * ignores any discovery `types` or `trusted-proxies` settings.
* If `requested-uri-discovery` is enabled (either because `enabled` is explicitly set to `true` or if either `types` or `trustedProxies` has been set), then Helidon performs a simple validity check: If `types` is specified and includes either `FORWARDED` or `X-FORWARDED`, then `trusted-proxies` must also be set explicitly. No `trusted-proxies` setting is needed if the user selects only the `HOST` discovery type.
* If discovery is enabled (explicitly or implicitly) and no discovery type is set, then Helidon defaults the type to `FORWARDED` and, therefore, requires at least an explicit `trusted-proxies` setting.